lib/memo_wise.rb

```diff
@@ -42,12 +41,13 @@ def #{method_name}
             end
           END_OF_METHOD
         else
+          # Note that we don't need to freeze args before using it as a hash key
+          # because Ruby always copies argument arrays when splatted.
           module_eval <<-END_OF_METHOD, __FILE__, __LINE__ + 1
             def #{method_name}(*args)
-              key = [:#{method_name}, args].freeze
-              @_memo_wise.fetch(key) do
-                @_memo_wise_keys[:#{method_name}] << args
-                @_memo_wise[key] = #{not_memoized_name}(*args)
+              hash = @_memo_wise[:#{method_name}]
```

Comment:
```suggestion
              hash = (@_memo_wise[:#{method_name}] ||= {})
```

What if we tried allocating a new Hash here?

Reply:
Wouldn't we still need to assign that hash to `@_memo_wise[:#{method_name}]`? My implementation looked something like:

```
@_memo_wise[:#{method_name}] = {} unless @_memo_wise.key?(:#{method_name})
hash = @_memo_wise[:#{method_name}]
...
```